source/extensions/quic_listeners/quiche/quic_filter_manager_connection_impl.cc

```diff
@@ -155,6 +155,7 @@ void QuicFilterManagerConnectionImpl::onConnectionCloseEvent(
     raiseConnectionEvent(source == quic::ConnectionCloseSource::FROM_PEER
                              ? Network::ConnectionEvent::RemoteClose
                              : Network::ConnectionEvent::LocalClose);
+    quic_connection_ = nullptr;
```

Comment:
What's the crash stack if not having this line? I suspect it fixes a reported crash in HCM doConnectionClose() if QUICHE closes its connection.

Reply:
AFIK this just moved nulling out the  quic_connection_ = nullptr; to make sure the connection event is consistently raised when we close the connection.
Unfortunately it looks like it's causing some tsan flake with the existing tests which I've been unable to reproduce localy.  I'm going to wait until your big fix lands, and see if it helps any, then update this PR again.